### PR TITLE
C++: Reduce path duplication

### PR DESCRIPTION
**Pull request by MathiasVP**
This fixes some issues uncovered in https://github.com/github/codeql/pull/11254.

### CODE REVIEW

File: cpp/ql/src/Security/CWE/CWE-311/CleartextTransmission.ql [hunk 1]
```diff
@@ -261,8 +261,7 @@ class ToEncryptionConfiguration extends TaintTracking2::Configuration {
   ToEncryptionConfiguration() { this = "ToEncryptionConfiguration" }
 
   override predicate isSource(DataFlow::Node source) {
-    any(FromSensitiveConfiguration config).hasFlow(source, _) and
-    isSourceImpl(source)
```
Comment (MathiasVP) [inline cpp/ql/src/Security/CWE/CWE-311/CleartextTransmission.ql:265]: Drive-by fix. This condition follows from `any(FromSensitiveConfiguration config).hasFlow(source, _)`.

File: cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll [hunk 2]
```diff
@@ -752,15 +752,26 @@ private predicate exprNodeShouldBeIndirectOperand(IndirectOperand node, Expr e,
...
+  instr = node.getOperand().getDef() and
+  not node instanceof ExprNode and
+  e = instr.getAst().(Expr).getUnconverted()
+}
+
 /** Holds if `node` should be an `IndirectOperand` that maps `node.asIndirectExpr()` to `e`. */
 private predicate indirectExprNodeShouldBeIndirectOperand(IndirectOperand node, Expr e) {
   exists(Instruction instr |
     instr = node.getOperand().getDef() and
     not node instanceof ExprNode
```
Comment (jketema) [inline cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll:767]: This now seems duplicated in the `indirectExprNodeShouldBeIndirectOperand0` predicate?

Comment (MathiasVP) [reply cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll:767]: True. I pulled it in there to reduce the size of `indirectExprNodeShouldBeIndirectOperand0`. Maybe that was a bit of a premature optimization 🤔. We want that condition to apply to all of `indirectExprNodeShouldBeIndirectOperand`, but we can't just have it in this new predicate because then it'll only apply to one side of the disjunct.

Comment (jketema) [reply cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll:767]: After some out-of-band discussion: we do need this in two places for the second disjunct, as `instr` needs to be restricted, and any other instruction that we capture `not indirectExprNodeShouldBeIndirectOperand0(_, _, e.getUnconverted())` also needs to satisfy it.

**Review by jketema — APPROVED** [review]
LGTM
